### PR TITLE
PathLike is not necessarily Path

### PR DESCRIPTION
**Pull request by cmwilhelm**
This was blowing up when a UPath was passed.

---

> [!NOTE]
> Use UPath and broaden accepted path types in model loading; add tests for path handling and HF download mocking.
> 
> - **model_loader**:
>   - Switch to `UPath` for artifact resolution, replacing `Path` and normalizing returned paths.
>   - Broaden path handling: `load_model_from_path` now accepts `PathLike | str`; `_resolve_artifact_path` accepts `ModelID | PathLike | str` and builds paths via `UPath`.
> - **Tests**:
>   - Add unit tests covering loading with `pathlib.Path`, `UPath`, and string paths; mocked HF downloads for `ModelID`; and `ModelID.repo_id()` format checks.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7e2557e459db2fc93c0321a128600004f4ef8c19. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: olmoearth_pretrain/model_loader.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def load_model_from_id(model_id: ModelID, load_weights: bool = True) -> torch.nn
 
 
 def load_model_from_path(
```
Comment (Hgherzog) [inline olmoearth_pretrain/model_loader.py:71]: Suggestion: add a test for this since it broke something down stream

File: olmoearth_pretrain/model_loader.py [hunk 2]
```diff
@@ -90,25 +90,27 @@ def load_model_from_path(
...
+    model_id_or_path: ModelID | PathLike[str], filename: str
+) -> PathLike:
     """Resolve the artifact file path for the specified model ID or path, downloading it from Hugging Face if necessary."""
     if isinstance(model_id_or_path, ModelID):
         return Path(
             hf_hub_download(repo_id=model_id_or_path.repo_id(), filename=filename)  # nosec
         )
     else:
-        return Path(model_id_or_path) / filename
+        return model_id_or_path / Path(filename)
```
[the remark this answers is not in the document]
Comment (Hgherzog) [reply olmoearth_pretrain/model_loader.py]: +1

File: olmoearth_pretrain/model_loader.py (hunk 1 above)
Comment (cmwilhelm) [reply olmoearth_pretrain/model_loader.py:71]: done
